http-svr/http_svr.ml

```diff
@@ -547,7 +547,8 @@ let read_body ?limit req bio =
 	| Some length ->
 		let length = Int64.to_int length in
 		maybe (fun l -> if length > l then raise Client_requested_size_over_limit) limit;
-		Buf_io.really_input_buf ~timeout:Buf_io.infinite_timeout bio length
+		if Buf_io.get_data bio = "" then Unixext.really_read_string (Buf_io.fd_of bio) length
```

Comment:
Well, `Buf_io.assert_buffer_empty` does it this way. If it's good enough for that then I assumed that it's good enough for this...?